### ISSUE TITLE
empty shared_vpc issue when using shared_vpc module

### ISSUE THREAD

**Issue by des-esseintes** [terraform-google-modules/terraform-google-project-factory#480]
Hey team,

I've faced the issue described here: https://github.com/terraform-google-modules/terraform-google-project-factory/issues/445#issuecomment-712709702 by @askoriy 
Basically shared_vpc module produces errors when shared_vpc is set to an empty value, it's attempting to create resources it should not, e.g. plan looks like:

```
resource "google_compute_shared_vpc_service_project" "shared_vpc_attachment" {
id              = (known after apply)
service_project = *******
}
```

this one is missing a required parameter `host_project`.

Not sure how to actually fix this tbh, I've checked the code and those resources should be omitted if shared_vpc == ""
But I've not spent too much time investigating this, tbh.

**Comment by bharathkkb** [comment]
@des-esseintes are you trying to create a regular project?

**Comment by des-esseintes** [comment]
not sure what's the definition of 'regular' here :)
i assume yes...
for now the workaround (quite an ugly one, imo) is to use the root module for the projects with no shared_vpcs, but...

**Comment by bharathkkb** [comment]
@des-esseintes apologies if I was not clear, what I meant was were you trying to create a project that is not a SVPC host/service project?
- If you use the `shared_vpc` module, it[ will create](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/65534493d95c284df04d0f9c46fb39814a32b9eb/modules/shared_vpc/main.tf#L40) a service project for you (note: [this requires](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/65534493d95c284df04d0f9c46fb39814a32b9eb/modules/core_project_factory/main.tf#L104) the `var. shared_vpc` to be set to the host project id)

- If you use the main module and explicitly set `enable_shared_vpc_host_project = true`, it will create a host project for you.

- Finally if you just want a regular project, you just use the main module, which by default creates a regular project.

**Comment by des-esseintes** [comment]
oh now i understand

so let me explain.

we had a module, which was using project factory as a submodule. we were on version 7.1. then i started migrating to 9.1 and found out that shared_vpcs no longer exist. after reading the comments in the ticket i mentioned in the topic start, i just changed module to shared_vpcs. that was ok for the projects with shared vpc defined, but started to fail for projects with an empty value.

i'm trying to find a better workaround rather than 2 separate modules, one of which would be using project factory and another one - shared vpcs submodule.

**Comment by bharathkkb** [comment]
@des-esseintes got it, we are trying converge with https://github.com/terraform-google-modules/terraform-google-project-factory/pull/446

**Comment by des-esseintes** [comment]
oh, thanks! i'll keep an eye on this one :)

**Comment by des-esseintes** [comment]
As this one seems to be merged now https://github.com/terraform-google-modules/terraform-google-project-factory/commit/0a6b9b9b6226484cba16a210f429817fd0d03aed, I think I'm going to test the feature when the next version of the module comes out and report the results here

**Comment by bharathkkb** [comment]
@des-esseintes you can also test the release branch now if you are interested. Closing this issue now, feel free to reopen if issue persists.
```hcl
source = "github.com/terraform-google-modules/terraform-google-project-factory?ref=release-v10.0.0"
```

**Comment by bharathkkb** [comment]
fixed by #446

**Comment by des-esseintes** [comment]
working as expected!
thanks!
